packages/yarnpkg-pnp/sources/esm-loader/loaderFlags.ts

```diff
@@ -20,3 +20,6 @@ export const SUPPORTS_IMPORT_ATTRIBUTES = major >= 21 || (major === 20 && minor
 
 // https://github.com/nodejs/node/pull/52104
 export const SUPPORTS_IMPORT_ATTRIBUTES_ONLY = major >= 22;
+
+// https://github.com/nodejs/node/pull/53725
+export const SUPPORTS_TYPE_STRIPPING = major > 23 || (major === 23 && minor > 6);
```

Comment:
There is a feature flag for this https://nodejs.org/docs/v22.14.0/api/process.html#processfeaturestypescript